src/V3Name.cpp

```diff
@@ -107,12 +107,6 @@ class NameVisitor final : public VNVisitor {
             iterateChildren(nodep);
         }
     }
-    void visit(AstMemberSel* nodep) override {
-        if (!nodep->user1()) {
-            rename(nodep, false);
-            iterateChildren(nodep);
-        }
-    }
```

Comment:
This simply suppresses the rename, and results in an inconsistent name:
```
  1:2:2:2:2:2: MEMBERSEL ...  queue
                 -> VAR ...  __PVT__queue

```

Instead I think this would work, see if you agree:

```
     void visit(AstMemberDType* nodep) override {
         if (!nodep->user1()) {
-            rename(nodep, false);
+            rename(nodep, true);
             iterateChildren(nodep);
         }
     }
     void visit(AstMemberSel* nodep) override {
         if (!nodep->user1()) {
+            rename(nodep, true);
+            iterateChildren(nodep);
+        }
+    }
+    void visit(AstStructSel* nodep) override {
+        if (!nodep->user1()) {
+            rename(nodep, true);
             iterateChildren(nodep);
         }
     }
```

Trying this I found a unrelated problem in structure naming, please rebase from master.

Reply:
I didn't notice that names are inconsistent. I have checked that the problem is present also on master.

I have a question about https://github.com/verilator/verilator/commit/3fc684359aaf3d5b3afafa3f73d3740961221b62. Why we started storing a name in AstMemberSel? I think that it should always be the same as the name of the variable that member select points to, so we could print the name of that variable instead of storing a copy. As we noticed, it doesn't work correctly and old name is printed, but maybe it would be better to fix it? Then I think we can remove visitors of select nodes on V3Name stage.

Reply:
I have checked and Your solution works. If we need to store names of selects, I think that it should be used instead of mine.

Reply:
I think the name is needed because before elaboration we don't have the variable and need to match them up, like VarRef's.

It would probably be good at some point to add a check to V3Broken to check for name consistency in all stages past linking.

Can you add "my" name change to this pull please, then on pass it can squash merge.